packages/rs-drive-abci/src/execution/validation/state_transition/check_tx_verification/v0/mod.rs

```diff
@@ -0,0 +1,188 @@
+                                );
+                        }
+                        Some(state_transition_action_result.into_data()?)
+                    }
+                } else {
+                    None
+                };
+
+                //
+                let result = state_transition.validate_identity_and_signatures(
+                    platform.drive,
+                    action.as_ref(),
+                    None,
+                    &mut state_transition_execution_context,
+                    platform_version,
+                )?;
+                // Validating signatures
+                if !result.is_valid() {
+                    return Ok(
+                        ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                            result.errors,
+                        ),
+                    );
+                }
+                let maybe_identity = result.into_data()?;
+
+                let action = if let Some(action) = action {
+                    action
+                } else {
+                    let state_transition_action_result =
+                        state_transition.transform_into_action(platform, true, None)?;
+                    if !state_transition_action_result.is_valid_with_data() {
+                        return Ok(
+                            ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                                state_transition_action_result.errors,
+                            ),
+                        );
+                    }
+                    state_transition_action_result.into_data()?
+                };
+
+                let execution_event = ExecutionEvent::create_from_state_transition_action(
+                    action,
+                    maybe_identity,
+                    platform.state.epoch_ref(),
+                    platform_version,
+                )?;
+
+                Ok(
+                    ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_data(Some(
+                        execution_event,
+                    )),
+                )
+            }
+        }
+        CheckTxLevel::Recheck => {
+            if let Some(asset_lock) = state_transition.asset_lock() {
+                // we should check that the asset lock is still valid
+                let validation_result =
+                    asset_lock.validate_state(platform, None, platform_version)?;
+                return if validation_result.is_valid() {
+                    Ok(ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_data(None))
+                } else {
+                    Ok(
+                        ConsensusValidationResult::<Option<ExecutionEvent>>::new_with_errors(
+                            validation_result.errors,
+                        ),
+                    )
+                };
+            } else {
+                let state_transition_action_result =
```

Comment:
Don't we need to perform all validation with dry run to validate balance?

Reply:
This is what is happening.

Reply:
Ok, we agreed I will write something here to explain why it's not here